lib/retry.js

```diff
@@ -64,16 +65,19 @@ export default function retry(times, task, callback) {
     var DEFAULT_TIMES = 5;
     var DEFAULT_INTERVAL = 0;
 
-
     var opts = {
         times: DEFAULT_TIMES,
-        interval: DEFAULT_INTERVAL
+        intervalFunc: constant(DEFAULT_INTERVAL)
     };
 
     function parseTimes(acc, t) {
         if (typeof t === 'object') {
             acc.times = +t.times || DEFAULT_TIMES;
-            acc.interval = +t.interval || DEFAULT_INTERVAL;
+
+            acc.intervalFunc = (typeof t.interval === 'function' ?
```

Comment:
extraneous bracket.

```js
acc.intervalFunc = typeof t.interval === 'function' ?
                t.interval :
                constant(+t.interval || DEFAULT_INTERVAL);
```

Otherwise looks good to me and +1